From 94f8bd7e551a88b675310b62f279d0730e6aefec Mon Sep 17 00:00:00 2001
From: Alynx Zhou <alynx.zhou@gmail.com>
Date: Fri, 5 Nov 2021 16:09:15 +0800
Subject: [PATCH] Revert "tracker: Use faster query in "tracker3 info""

This reverts commit 0c569af986c0038a865793b7fdecc054a427c76b,
which fails to show metadata for some files.
---
 src/tracker/tracker-info.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/tracker/tracker-info.c b/src/tracker/tracker-info.c
index 72f71ee59..a072ec8f8 100644
--- a/src/tracker/tracker-info.c
+++ b/src/tracker/tracker-info.c
@@ -525,9 +525,13 @@ info_run (void)
 		query = g_strdup_printf ("SELECT DISTINCT ?predicate ?object ?x"
 		                         "  ( EXISTS { ?predicate rdfs:range [ rdfs:subClassOf rdfs:Resource ] } )"
 		                         "WHERE {"
-		                         "  <%s> nie:interpretedAs? ?x . "
-					 "  ?x ?predicate ?object . "
-		                         "} ORDER BY ?x",
+		                         "  { <%s> ?predicate ?object . } "
+					 "  UNION "
+					 "  { <%s> nie:interpretedAs ?x . "
+					 "    ?x ?predicate ?object . "
+	                                 "  } "
+		                         "}",
+					 urn,
 					 urn);
 
 		cursor = tracker_sparql_connection_query (connection, query, NULL, &error);
-- 
2.33.1

